lib/roby/app/scripts/test.rb

```diff
@@ -101,20 +110,25 @@
     begin
         Roby.app.prepare
 
+        def discover_test_files(all:, only_self:, base_dir:)
+            self_files, dependent_files = \
+                Roby.app.discover_test_files(
+                    all: all, only_self: only_self, base_dir: base_dir
+                ).map(&:first).partition { |f| Roby.app.self_file?(f) }
+
+            self_files.sort + dependent_files.sort
+        end
+
+        # list_tests = true
         if test_files.empty?
-            test_files = app.discover_test_files(
-                all: all, only_self: only_self
-            ).map(&:first)
-            self_files, dependent_files =
-                test_files.partition { |f| app.self_file?(f) }
-            test_files = self_files.sort + dependent_files.sort
+            test_files = discover_test_files(all: all, only_self: only_self, base_dir: base_dir)
             if list_tests
                 puts "Would load #{test_files.size} test files"
                 test_files.each do |path|
                     puts "  #{path}"
                 end
 
-                all_existing_tests = app.find_dirs("test", order: :specific_first, all: !only_self).inject(Set.new) do |all, dir|
+                all_existing_tests = Roby.app.find_dirs("test", order: :specific_first, all: !only_self).inject(Set.new) do |all, dir|
```

Comment:
why `#inject` was used here instead of `#each_with_objects`

Reply:
Probably because when I wrote that code I either did not know about each_with_object, or it did not exist yet.